src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs

```diff
@@ -149,19 +200,38 @@ public static void Multiply(ReadOnlySpan<uint> left, uint right, Span<uint> bits
             bits[i] = (uint)carry;
         }
 
-#if DEBUG
-        // Mutable for unit testing...
-        internal static
-#else
-        internal const
-#endif
-            int MultiplyKaratsubaThreshold = 32;
+        /// <summary>
+        /// A wrapper of <see cref="MultiplyImpl(ReadOnlySpan{uint}, ReadOnlySpan{uint}, Span{uint})"/>.
+        /// </summary>
+        /// /// <remarks>
+        /// The order of <paramref name="left"/> and <paramref name="right"/> does not matter.
+        /// The method internally swaps them if necessary to ensure correct computation.
+        /// </remarks>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static void Multiply(ReadOnlySpan<uint> left, ReadOnlySpan<uint> right, Span<uint> bits)
+        {
+            if (left.Length < right.Length)
+            {
+                if (left.Length > 0)
+                    MultiplyImpl(right, left, bits);
+            }
+            else
+            {
+                if (right.Length > 0)
+                    MultiplyImpl(left, right, bits);
+            }
```

Comment:
This method does not _clear_ the `bits` to all zero (that would seem to be what the result of multiplying a zero-length multiplicand). This is a public method, so I'm concern about folks having expectations that are not documented/met.

Questions:

1. Is my assumption that either multiplicand having a zero `.Length` implies it's actually value-zero?
2. Do we need to clear the output `bits` or set it's length or anything?
3. Would this logic be clearer?

```c#
 if (left.Length != 0 && right.Length != 0)
 {
     if (left.Length < right.Length)
     {
         MultiplyImpl(right, left, bits);
     }
     else
     {
         MultiplyImpl(left, right, bits);
     }
}
// bits.Clear(); // needed?
```

Reply:
The `BigIntegerCalculator` class is internal, so it is sufficient to use `Debug.Assert` to ensure that the `bits` is filled with zeros.

This assertion already exists in main.
https://github.com/dotnet/runtime/blob/77115c33ec4fb8688e4803cd8f76cd0fa4bff7d7/src/libraries/System.Runtime.Numerics/src/System/Numerics/BigIntegerCalculator.SquMul.cs#L163

Reply:
1. If `right.Length` is zero, it means that `bits` remains unchanged. Given the assumption that `bits` is zero-filled, this is equivalent to multiplying by zero.
2. This function is called recursively, so invoking `.Clear()` should be avoided in the function. It is wasted except for the first call.
3. Currently, only two comparisons are needed: `left.Length < right.Length` and whether `left.Length > 0` or `right.Length > 0`. However, with your approach, three comparisons are required.

Reply:
Great! Thanks for the explainer :)